### ISSUE TITLE
Unhelpful ValueError raised when color array is incorrect format

### ISSUE THREAD

**Issue by TTitcombe** [matplotlib/matplotlib#15727]
### Bug report

**Bug summary**
When colors passed to a plot are provided as tuples in the range `[0, 255]` instead of `[0, 1]`, the error raised relates to the shape of the data (which is correct), not the values of the data.

**Code for reproduction**

```python
import matplotlib.pyplot as plt

plt.scatter([0, 1], [0, 1], c=[(255, 0, 0), (0, 255, 0)])
```

**Actual outcome**

```python
ValueError: 'c' argument has 2 elements, which is not acceptable for use with 'x' with size 2, 'y' with size 2.
```

_Edit_:
```python
ValueError: RGBA values should be within 0-1 range
```
is also raised, however it is less prominent than the "size" error, which should not be raised at all

**Expected outcome**
A single, specific error message:
```python
ValueError: RGBA values should be within 0-1 range
```

**Matplotlib version**
  * Operating system: Windows 10
  * Matplotlib version: 3.1.2
  * Matplotlib backend: Qt5Agg
  * Python version: 3.7.3

Installed matplotlib from conda

**Comment by TTitcombe** [comment]
The correct error is being raised in from [`colors = mcolors.to_rgba_array(c)`](https://github.com/matplotlib/matplotlib/blob/master/lib/matplotlib/axes/_axes.py#L4266) in `_axes.py`. However, this is caught in `_axes.py` and the incorrect `ValueError` is raised [here](https://github.com/matplotlib/matplotlib/blob/master/lib/matplotlib/axes/_axes.py#L4269). 

The incorrect error is raised because `valid_shape` is `False`, which is being set [here](https://github.com/matplotlib/matplotlib/blob/master/lib/matplotlib/axes/_axes.py#L4263). This is caused by `c.size == xsize` evaluating to `False` when `c` is an array of tuples because `c.size` will be `number of tuple X size of tuple`, so for `c=[(255, 0, 0), (0, 255, 0)]`, `c.size == 6` when that statement is intended to check if it is `2`.

I haven't had the time to work out exactly what the intention of this bit of the code is, so have not yet found a solution.

**Comment by QuLogic** [comment]
This was fixed by #17245 in 3.3.0.
